app/features/reminder.py

```diff
@@ -0,0 +1,182 @@
+import asyncio
+import datetime
+import zoneinfo
+
+import interactions
+
+
+class Alarm(interactions.Extension):
+    """Alarm / Reminder extension."""
+
+    def __init__(self, bot: interactions.Client) -> None:
+        self.bot = bot
+        print(f"{self.bot.user} loaded alarm extension")
+        self.tz = UserTimezones()
+
+    base = interactions.SlashCommand(name="remindme", description="Alarm base group")
+    set = interactions.SlashCommand(name="set", description="Set base group")
+
+    def _base_in_opts() -> list[interactions.SlashCommandOption]:
+        """Return options for the base in subcommand."""
+        return [
+            interactions.SlashCommandOption(
+                name="duration",
+                description="integer representing the number of units",
+                required=True,
+                type=interactions.OptionType.INTEGER,
+            ),
+            interactions.SlashCommandOption(
+                name="units",
+                description="units representing a measurement of time",
+                required=True,
+                type=interactions.OptionType.STRING,
+                choices=[
+                    interactions.SlashCommandChoice(name="Seconds", value="sec"),
+                    interactions.SlashCommandChoice(name="Minutes", value="min"),
+                    interactions.SlashCommandChoice(name="Hours", value="hour"),
+                ],
+            ),
+            interactions.SlashCommandOption(
+                name="message",
+                description="message to be sent when the alarm is triggered",
+                required=False,
+                type=interactions.OptionType.STRING,
+            ),
+        ]
+
+    def _base_at_opts() -> list[interactions.SlashCommandOption]:
+        """Return options for the base at subcommand."""
+        return [
+            interactions.SlashCommandOption(
+                name="date_time",
+                description="date and time in the format 'YYYY-MM-DD HH:MM'",
+                required=True,
+                type=interactions.OptionType.STRING,
+            ),
+            interactions.SlashCommandOption(
+                name="message",
+                description="message to be sent when the alarm is triggered",
+                required=False,
+                type=interactions.OptionType.STRING,
+            ),
+        ]
+
+    @base.subcommand(
+        sub_cmd_name="at",
+        sub_cmd_description="add an alarm at a specific date and time",
+        options=_base_at_opts(),
+    )
+    async def add_at(
+        self,
+        ctx: interactions.SlashContext,
+        date_time: str,
+        message: str = "Your previously set reminder has been triggered",
+    ) -> None:
+        """Create a reminder for a specific time of the day based on a time input."""
+        if not self.tz.has_user(ctx.member):
```

Comment:
The passed argument doesn't have the same type as typehinted in the method.
IDEs can highlight things like that, to help avoid making mistakes.
![image](https://github.com/user-attachments/assets/8ecb202e-28e5-4510-bc48-ffb0a5c37294)
If a 'Member' parameter is passen, then the 'Member' should also be the one expected in the typehint.